src/bootstrap/setup.rs

```diff
@@ -219,6 +229,38 @@ fn try_link_toolchain(stage_path: &str) -> bool {
         .map_or(false, |output| output.status.success())
 }
 
+fn create_stage1_toolchain_placeholder_if_it_does_not_already_exist(stage_path: &str) -> bool {
+    let mut pathbuf = PathBuf::from(stage_path);
+
+    pathbuf.push("lib");
+    if fs::create_dir_all(&pathbuf).is_err() {
+        return false;
+    };
+    pathbuf.pop();
```

Comment:
Rather than push/popping, I suspect a .join(...) may work essentially just as well.

Reply:
I borrowed code from https://github.com/rust-lang/rustup/blob/5225e87a5d974ab5f1626bcb2a7b43f76ab883f0/src/toolchain.rs#L479-L497. I will change it to use `.join()` instead.

Reply:
Updated